pkg/apis/core/validation/shoot.go

```diff
@@ -1046,12 +1049,31 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 			allErrs = append(allErrs, field.Duplicate(fldPath.Index(i).Child("name"), worker.Name))
 		}
 		workerNames[worker.Name] = true
+
+		if len(worker.Taints) == 0 {
+			atLeastOnePoolWithCompatibleTaints = true
+		} else {
+			onlyPreferNoScheduleEffectTaints := true
+			for _, taint := range worker.Taints {
+				if taint.Effect != corev1.TaintEffectPreferNoSchedule {
+					onlyPreferNoScheduleEffectTaints = false
+					break
+				}
+			}
+			if onlyPreferNoScheduleEffectTaints {
+				atLeastOnePoolWithCompatibleTaints = true
+			}
+		}
```

Comment:
This block can go inside the following `if` check, right?
```
if atLeastOnePoolWithCompatibleTaints != true
```
Basically no need to process all worker groups if we have found atleast 1 favourable worker group.

Reply:
done